app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,18 @@
+module SessionsHelper
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
```

Comment:
tìm hiểu sự khác nhau giữa `present?` `nil?` và `blank?`

ở đây dùng `present?` để khỏi phải dùng nil? và nghịch đảo giữ liệu

Reply:
1. .nil? và .empty? là hàm của Ruby
    .blank và .present? là các hàm được thêm trong Rails: có nhiều ưu điểm hơn nên trong rails thường dùng 2 thèn này hơn.
2. nil? là 1 hàm của Object nên chỉ có nil.nil? trả về true
3. empty? function có sẵn của String, Array and Hash. Nên nếu lenght của mấy ông này == 0 thì trả về true.
4. blank? một object được coi là blank nếu như nó false, empty hoặc là 1 chuỗi chỉ gồm các khoảng trắng.
    đối với Array và Hash thì .blank? sẽ trả về true nếu nó empty.
    đối với NilClass, FalseClass thì .blank? trực tiếp trả về false
    các lớp TrueClass, Numeric thì .blank? trực tiếp trả về true
    đối với String: "\t\n\r\s      \u".blank? => true
5. present? Nếu 1 object không blank, vậy nó present.